stream/src/main/scala/org/apache/pekko/stream/javadsl/Flow.scala

```diff
@@ -4628,15 +4628,18 @@ final class Flow[In, Out, Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends Gr
   def aggregateWithBoundary[Agg, Emit](allocate: java.util.function.Supplier[Agg],
       aggregate: function.Function2[Agg, Out, Pair[Agg, Boolean]],
       harvest: function.Function[Agg, Emit],
-      emitOnTimer: Pair[java.util.function.Predicate[Agg], java.time.Duration]): javadsl.Flow[In, Emit, Mat] =
+      emitOnTimer: Optional[Pair[java.util.function.Predicate[Agg], java.time.Duration]])
```

Comment:
Is there any way that we could try method overloading? Keep the existing aggregateWithBoundary function and add an extra aggregateWithBoundary function that uses `Optional<Pair<???>>`.

Reply:
Yes, but I think that's unneeded, because,  the api is marked `ApiMayChange`

Reply:
we could deprecate the old version and move the ApiMayChange to the new version